bundles/org.openhab.binding.comfoair/src/main/java/org/openhab/binding/comfoair/internal/datatypes/DataTypeTemperature.java

```diff
@@ -0,0 +1,73 @@
+/**
+ * Copyright (c) 2010-2020 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.comfoair.internal.datatypes;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+import org.eclipse.smarthome.core.library.types.DecimalType;
+import org.eclipse.smarthome.core.library.types.QuantityType;
+import org.eclipse.smarthome.core.library.unit.SIUnits;
+import org.eclipse.smarthome.core.types.State;
+import org.eclipse.smarthome.core.types.UnDefType;
+import org.openhab.binding.comfoair.internal.ComfoAirCommandType;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Class to handle temperature values
+ *
+ * @author Holger Hees - Initial Contribution
+ * @author Hans Böhm - QuantityTypes
+ */
+@NonNullByDefault
+public class DataTypeTemperature implements ComfoAirDataType {
+    private static DataTypeTemperature SINGLETON_INSTANCE = new DataTypeTemperature();
+
+    private DataTypeTemperature() {
+    }
+
+    private final Logger logger = LoggerFactory.getLogger(DataTypeTemperature.class);
+
+    public static DataTypeTemperature getInstance() {
+        return SINGLETON_INSTANCE;
+    }
+
+    @Override
+    public State convertToState(int @Nullable [] data, ComfoAirCommandType commandType) {
+        if (data == null) {
+            logger.trace("\"DataTypeTemperature\" class \"convertToState\" method parameter: null");
+            return UnDefType.NULL;
+        } else {
+            int[] get_reply_data_pos = commandType.getGetReplyDataPos();
+            if (get_reply_data_pos != null && get_reply_data_pos[0] < data.length) {
+                return new QuantityType<>((((double) data[get_reply_data_pos[0]]) / 2) - 20, SIUnits.CELSIUS);
+            } else {
+                return UnDefType.NULL;
+            }
+        }
+    }
+
+    @Override
+    public int @Nullable [] convertFromState(State value, ComfoAirCommandType commandType) {
+        if (((DecimalType) value) instanceof DecimalType) {
```

Comment:
How come `convertToState` returns a `QuantityType` but this method expects a `DecimalType`?

Reply:
All `QuantityType` commands ar transformed to `DecimalType` in the [handler](https://github.com/openhab/openhab-addons/pull/7052/files#diff-5ea2270a1fc35f2c479ea7dac0b38fe6R190-R208). Therefore, it should be save, if an item is defined as `Number` instead of `Number:Temperature`.

Reply:
That is ok...I guess, but think it is a bit misleading when looking at the code. I think it would be better to remove  the `commandToState` and put all that logic into the individual ComfoAirDataType classes.

Reply:
The `convertToState` and `convertFromState` classes should be symmetrical in behavior.

Reply:
Moved the logic to ComfoAirCommandType and removed some redundant code on the Handler.